feat/feature.go

```diff
@@ -105,6 +105,24 @@ type Collection interface {
 	Location() Feature
 }
 
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the deepest non nil feature. The reference feature can be used by
+// different calls to BasePositionOf to verify that returned positions are
+// comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the parent feature of the swallowest nil feature. The reference feature
+// can be used by different calls to BasePositionOf to verify that returned
+// positions are comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
 // BasePositionOf returns the position in f converted to coordinates in the
```

Comment:
You have to remember thought that chromosome has a `Start()` of each own which might not be 0. Note that the definition of the chromosome's `Start()` is: the chromosome's start position relative to its location (which is nil). So basically `BasePositionOf()` for a feature on the chromosome returns the coordinates relative to the `nil`. In other words `BasePositionOf` will consume the chromosomes `Start()`. If we don't do this, then we have to accept that the coordinates of the deepest non-nil feature have no effect and are always ignored.